tests/testthat/test-drive_list.R

```diff
@@ -8,97 +8,133 @@ context("List files")
 # (pesky_files <- drive_list(pattern = "foo|bar|baz|yo"))
 # pesky_files$id %>% purrr::map(drive_file) %>% purrr::map(drive_delete)
 
+nm_ <- function(x) paste0(x, "-TEST-drive-list")
+
+run <- FALSE
+clean <- FALSE
+if (run) {
+  ## make sure directory is clean
+  if (clean) {
+    del_ids <- drive_list(pattern = paste(c(nm_("foo"),
+                                            nm_("foo2"),
+                                            nm_("bar"),
+                                            nm_("baz"),
+                                            nm_("yo")),
+                                          collapse = "|")
+    )$id
+    del_files <- purrr::map(del_ids, drive_file)
+    del <- purrr::map(del_files, drive_delete, verbose = FALSE)
+  }
+
+  ## drive_list() not confused by same-named folders
+  ## get_leaf() errors when two distinct folders have same path
+  ## +-- foo
+  ## | +-- bar
+  ## | +-- bar
+  ## | | +-- baz
+  ## | | | +-- yo
```

Comment:
Do you think it would be useful to repeat these directory trees in the associated tests?